include/gsl/gsl_byte

```diff
@@ -37,10 +37,30 @@
 #pragma push_macro("noexcept")
 #define noexcept /*noexcept*/
 #endif           // _MSC_VER <= 1800
+
+#if _HAS_STD_BYTE 
+
```

Comment:
@neilmacintosh I understand your conundrum, but I think on average the compiler version is the better version.  The feature test macro `_HAS_STD_BYTE` is a last resort escape hatch and may go away (at least that has been my understanding from internal discussion with libs folks.)